src/huggingface_hub/hf_api.py

```diff
@@ -606,9 +618,66 @@ def list_models(
         d = r.json()
         return [ModelInfo(**x) for x in d]
 
+    def _unpack_model_filter(self, model_filter: ModelFilter):
+        """
+        Unpacks a `ModelFilter` into something readable for `list_models`
+        """
+        model_str = ""
+        tags = []
+
+        # Handling author
+        if model_filter.author is not None:
+            model_str = f"{model_filter.author}/"
+
+        # Handling model_name
+        if model_filter.model_name is not None:
+            model_str += model_filter.model_name
+
+        filter_tuple = []
+
+        # Handling tasks
+        if model_filter.task is not None:
+            if not isinstance(model_filter.task, (list, tuple)):
+                model_filter.task = [model_filter.task]
+            for task in model_filter.task:
+                filter_tuple.append(task)
+
+        # Handling dataset
+        if model_filter.trained_dataset is not None:
+            if not isinstance(model_filter.trained_dataset, (list, tuple)):
+                model_filter.trained_dataset = [model_filter.trained_dataset]
+            for dataset in model_filter.trained_dataset:
+                if "dataset:" not in dataset:
+                    dataset = f"dataset:{dataset}"
+                filter_tuple.append(dataset)
+
+        # Handling framework
+        if model_filter.framework:
+            if not isinstance(model_filter.framework, (list, tuple)):
+                model_filter.framework = [model_filter.framework]
+            for framework in model_filter.framework:
+                filter_tuple.append(framework)
+
+        # Handling tags
+        if model_filter.tags:
+            if not isinstance(model_filter.tags, (list, tuple)):
+                tags.append(model_filter.tags)
+            else:
+                tags += model_filter.tags
```

Comment:
I think all of this can be simplified to `tags.extend` as well and then there's no need to do the type checking

Reply:
And also https://github.com/huggingface/huggingface_hub/pull/553/files#diff-71a3e23532e1355e79edd51b351bd514f0aa55b946e4c530b0c95f647773ac5dR640-R643

Reply:
Some of it can be, but in some cases we check to make sure the string tag is there (such as `dataset:`), so in some cases it should stay

Reply:
Since a string is an array of char, it's not quite as simple as that. I did refactor it to just be a type check and extend either the value or convert to a list, but simply using `.extend` alone doesn't work.